src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -499,6 +499,53 @@ internal void Insert(ref T[] list, string? name, MemberListType listType)
                     }
                 }
 
+                private void MergeWithGlobalListInOrder(T[] list)
+                {
+                    T[]? cachedMembers = m_allMembers;
+
+                    if (cachedMembers == null)
+                    {
+                        m_allMembers = list;
+                        return;
+                    }
+
+                    int length = list.Length;
+                    T[] newCache = list;
+
+                    // When all members are requested there should be no element in cache that not found within all members list.
+                    // But somehow in very rare case such element found in cache and not appending that to the all members list
+                    // causing random test failure, must be related to the bug commented on row 597
+                    Array.Resize(ref newCache, length + 1);
+
+                    foreach (T cachedMemberInfo in cachedMembers)
+                    {
+                        bool foundInList = false;
+
+                        if (cachedMemberInfo == null)
+                            break;
+
+                        for (int i = 0; i < list.Length; i++)
+                        {
+                            T newMemberInfo = list[i];
+
+                            if (newMemberInfo.CacheEquals(cachedMemberInfo))
+                            {
+                                newCache[i] = cachedMemberInfo;
+                                list[i] = cachedMemberInfo;
+                                foundInList = true;
+                                break;
+                            }
+                        }
+
+                        if (!foundInList)
+                        {
+                            newCache[length] = cachedMemberInfo;
```

Comment:
Doesn't this only handle 1 "not found" item?

Reply:
Yes, if there is one found. In case you did not notified I have created [PR](https://github.com/dotnet/runtime/pull/69506) in dotnet/runtime with this